L1Trigger/L1TCalorimeter/interface/AccumulatingSort.h

```diff
@@ -0,0 +1,50 @@
+#include <vector>
+#include <list>
+#include <stdint.h>
+
+
+template < typename T >
+class AccumulatingSort
+{
+
+private:
+    std::vector < std::list < T > > mSortArrays;
+    uint32_t mSize;
+
+private:
+    void AccumulatorUnit( const std::list< T >& aInput , T& aAcc , std::list< T >& aTail )
+    {
+        aTail.clear();
+
+        bool lAccInserted( false );
+
+        for( typename std::list< T >::const_iterator lIt( aInput.begin() ) ; lIt != aInput.end() ; ++lIt )
```

Comment:
does

aTail.reserve(aInput.size()) 

not work?

> On Sep 6, 2016, at 1:30 PM, rekovic <notifications@github.com> wrote:
> 
> In L1Trigger/L1TCalorimeter/interface/AccumulatingSort.h:
> 
> > +template < typename T >
> > +class AccumulatingSort
> > +{
> > +
> > +private:
> > +    std::vector < std::list < T > > mSortArrays;
> > +    uint32_t mSize;
> > +
> > +private:
> > +    void AccumulatorUnit( const std::list< T >& aInput , T& aAcc , std::list< T >& aTail )
> > +    {
> > +        aTail.clear();
> > +
> > +        bool lAccInserted( false );
> > +
> > +        for( typename std::list< T >::const_iterator lIt( aInput.begin() ) ; lIt != aInput.end() ; ++lIt )
> 
> @davidlange6 Since lists don't have method, I am not sure what exactly you meant here with reserving the memory. Method would value-initilize the list. Did you have something particular in mind?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>